MetaMask/metamask-extension
# fix: reduce size of token list fetch

## **Description**

When switching to a network for the first time, MM checks whether a token list is available for the network.  If not, the 3rd item is shown here:

<img width="317" alt="Screenshot 2024-04-30 at 10 29 29 AM" src="https://github.com/MetaMask/metamask-extension/assets/3500406/5cbe1d3c-8d8d-4b81-bc0a-2a4e492cbae0">

But we don't need the entire token list, since we're only checking whether a list is available.  Adding query parameters to reduce the response size.  On mainnet for example, 1.5MB (gzipped) to < 1KB.


[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/24316?quickstart=1)

## **Related issues**


## **Manual testing steps**

1. Open metamask
2. Visit a chain that has token lists, like linea
3. The popup should not warn about missing tokens
4. Visit a chain that does not have token lists, like filecoin
5. The popup should warn about missing tokens

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
